gt/main.c

```diff
@@ -705,55 +706,62 @@ alloc_and_fill_notify_pkt(unsigned int socket, struct ggu_policy *policy,
 
 	/* Fill up the policy decision. */
 	memset(notify_ggu, 0, sizeof(*notify_ggu));
-	notify_ggu->v1 = GGU_PD_VER1;
+	notify_ggu->version = GGU_PD_VER;
 	if (policy->flow.proto == ETHER_TYPE_IPv4
 			&& policy->state == GK_DECLINED) {
-		notify_ggu->n1 = 1;
-		data = (uint8_t *)rte_pktmbuf_append(notify_pkt,
-			sizeof(policy->flow.f.v4) +
-			sizeof(policy->params.u.declined));
-		rte_memcpy(data, &policy->flow.f.v4,
+		decision_len = sizeof(policy->flow.f.v4) +
```

Comment:
Why doesn't `decision_len` include `sizeof(struct ggu_decision)`? If it's a bug, we need to find all the places that are affected, and there're side effects for the next patch as well.